proposals/csharp-9.0/function-pointers.md

```diff
@@ -268,6 +268,75 @@ The better function member specification will be changed to include the followin
 
 This means that it is possible to overload on `void*` and a `delegate*` and still sensibly use the address-of operator.
 
+### Type Inference
+
+In unsafe code, the following changes are made to the type inference algorithms:
+
+#### Input types
+
+https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#input-types
+
+The following is added:
+
+> If `E` is an address-of method group and `T` is a function pointer type then all the parameter types of `T` are input types of `E` with type `T`.
+
+#### Output types
+
+https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#output-types
+
+The following is added:
+
+> If `E` is an address-of method group and `T` is a function pointer type then the return type of `T` is an output type of `E` with type `T`.
+
+#### Output type inferences
+
+https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#output-type-inferences
+
+The following bullet is added between bullets 2 and 3:
+
+> * If `E` is an address-of method group and `T` is a function pointer type with parameter types `T1..Tk` and return type `Tb`, and overload resolution
+of `E` with the types `T1..Tk` yields a single method with return type `U`, then a _lower-bound inference_ is made from `U` to `Tb`.
+
+#### Lower-bound inferences
+
+https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#lower-bound-inferences
+
+The following case is added to bullet 3:
+
+> * `V` is a function pointer type `delegate*<V2..Vk, V1>` and there is a unique type `delegate*<U2..Uk, U1>` such that `U` (or, if `U` is a type parameter,
+it's effective base class or any member of its effective interface set) is identical to, inherits from (directly or indirectly), implements (directly or
+indirectly), or has an implicit function pointer conversion to `delegate*<U2..Uk, U1>`.
```

Comment:
> unique type `delegate*<U2..Uk, U1>` such that `U` (or, if `U` is a type parameter,
it's effective base class or any member of its effective interface set) is identical to, inherits from (directly or indirectly), implements (directly or
indirectly)

I think almost of this is applicable to a function pointer type:
- it is not a class, thus cannot be inherited from or be an effective base class
- it is not an interface, thus cannot be implemented

If I remember correctly, pointers cannot be type parameters. If that is the case, probably no need to deal with constraints at all. And we will be left with just: "`delegate*<U2..Uk, U1>` such that `U` is identical to"

Reply:
> `delegate*<U2..Uk, U1>` such that `U` is identical to

I believe this needs to be "`delegate*<U2..Uk, U1>` such that `U` has an implicit function pointer conversion to", because we want to perform lower-bound inference from each Vi to Ui, not exact inference.